agent/engine/docker_task_engine.go

```diff
@@ -1015,8 +1015,12 @@ func (engine *DockerTaskEngine) GetTaskByArn(arn string) (*apitask.Task, bool) {
 
 func (engine *DockerTaskEngine) pullContainer(task *apitask.Task, container *apicontainer.Container) dockerapi.DockerContainerMetadata {
 	switch container.Type {
-	case apicontainer.ContainerCNIPause, apicontainer.ContainerNamespacePause:
-		// pause images are managed at startup
+	case apicontainer.ContainerCNIPause, apicontainer.ContainerNamespacePause, apicontainer.ContainerServiceConnectRelay:
+		// pause images and AppNet relay image are managed at startup
+		return dockerapi.DockerContainerMetadata{}
+	}
+	// AppNet Agent container image is also managed at start up (it uses the same image as AppNet Relay container)
+	if task.IsServiceConnectEnabled() && container == task.GetServiceConnectContainer() {
```

Comment:
would it not be easier to just make the containerType of the ServiceConnectContainer within the task be of type apicontainer.ContainerServiceConnectRelay ? then it is handled in the previous switch block.

Reply:
Synced offline - there are a few places where we use container type `ContainerServiceConnectRelay` to specify relay-only behavior, which are not applicable to task AppNet containers.